src/der.rs

```diff
@@ -222,3 +226,163 @@ macro_rules! oid {
         [(40 * $first) + $second, $( $tail ),*]
     )
 }
+
+#[cfg(feature = "alloc")]
+pub(crate) fn wrap_in_asn1_len(bytes: &mut Vec<u8>) {
```

Comment:
To support RFC 5915 Sec1 encoded P-256/P-384 keys this file is now updated with some helpers taken from Rustls, converted to avoid using anything Rustls-specific

* https://github.com/rustls/rustls/blob/main/rustls/src/x509.rs
* https://github.com/rustls/rustls/blob/4d710591c134ee1dbc8ed9715354f3e29709beec/rustls/src/sign.rs#L289-L336

In a perfect world all of this code would go into Ring (see https://github.com/briansmith/ring/pull/1456), but I don't think that will happen anytime soon.

Open questions:
* The more aggressive `clippy` config used by `webpki` required me to allow some `as_conversions`. I've annotated these and indicated why I think they're safe here but would appreciate a double check. Otherwise we should convert these to use `try_Into` and make the functions falliable.

* I haven't exported these helpers for use outside of webpki. I can do that, but so far the webpki `der` module has been crate private and so I wasn't sure if everyone is on-board with this visibility change. It makes sense to me that we'd want to remove the duplicate code from Rustls to use the webpki versions, so perhaps we should make the visbillity change?

Reply:
This function as specified is a little ugly (I know this the formulation used in rustls today -- tried to optimize it a bit recently and decided to postpone that) because it has to effectively copy the data once for every byte needed to represent the length. It would be nice to reformulate `wrap_in_sequence()`/`wrap_in_asn1_len()` to do something more efficient.

(Ideally if we know the length of the data separately from having access to the buffer the full sequence is about to be encoded into, we would write the sequence tag, the length bytes, and then the data. If the target buffer is the one the data currently lives in, ideally we'd only copy the data once instead of 2 + log n times. For both of these cases, we could make sure that we don't reallocate multiple times while growing the destination buffer.)

(Also as previously discussed, I would usually put these little helper functions below the more substantial, higher-level, more public API in the module.)

Reply:
Is https://github.com/rustls/webpki/commit/b8d82e583184d38790364787ae898e87490e67f0 any closer to what you had in mind? If so I'll fix it up and fold back into this branch.

Reply:
@djc How about something like https://github.com/rustls/webpki/commit/fb0cc1424c03b49afc913975e0919e4a79e5be84 

Two main differences;
* I'm allocating one Vec up-front that should hold the entire result without needing to expand itself.
* To avoid shifting the data around, I'm building the Vec contents by appending data into the Vec and then flipping it all at the end.

It's a little bit fiddly working with variable length DER encodings and I'm not sure my very specialized "write a DER length but in reverse" helper is up to snuff, but I haven't been able to think of another way to achieve a result that avoids shifting the data around or allocating more than appropriate. 

I might need to pair with you on a different approach if this version isn't workable (or maybe we could land the feature with the Rustls formulation and then iterate on optimization separately?)

LMK what you think :-)